### PR TITLE
fix: Restore the MVCC functionality.

### PR DESCRIPTION
**Pull request by czs007**
When the TimeTravel functionality was previously removed, it inadvertently affected the MVCC functionality within the system. This PR aims to reintroduce the internal MVCC functionality as follows:

 1. Add MvccTimestamp to the requests of Search/Query and the results of Search internally.
 2. When the delegator receives a Query/Search request and there is no MVCC timestamp set in the request, set the delegator's current tsafe as the MVCC timestamp of the request. If the request already has an MVCC timestamp, do not modify it.
 3. When the Proxy handles Search and triggers the second phase ReQuery, divide the ReQuery into different shards and pass the MVCC timestamp to the corresponding Query requests.

issue: #29656

### CODE REVIEW

File: internal/querynodev2/segments/result.go [hunk 1]
```diff
@@ -49,6 +49,12 @@ func ReduceSearchResults(ctx context.Context, results []*internalpb.SearchResult
 		return results[0], nil
```
Comment (bigsheeper) [inline internal/querynodev2/segments/result.go:49]: did not set `channelsMvcc` in this case

**Comment by samhuang-z** [comment]
rerun ut

**Pull request merged**